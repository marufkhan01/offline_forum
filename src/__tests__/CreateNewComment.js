import React from 'react';
import { shallow, mount} from 'enzyme';
import toJSON from 'enzyme-to-json';
import CreateNewComment from '../components/CreateNewComment';

describe('create comment', () => {
    const wrapper = mount(<CreateNewComment postId="1" author='Morgana' updateComments={() => {}}/>);

    it('create new comment', () => {
        wrapper.find('textarea[name="comment"]').simulate('change', {
            target: {name: 'comment', value: 'Morgana comment'}
        });
        expect(wrapper.state().comment).toBe('Morgana comment');
    });


    it('CreateNewComment component should render as expected', () => {
        const component = shallow(<CreateNewComment/>);
        const tree = toJSON(component);
        expect(tree).toMatchSnapshot();
    });
});