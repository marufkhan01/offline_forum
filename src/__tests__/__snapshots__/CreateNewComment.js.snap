// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateNewComment component should render as expected 1`] = `
<form
  className="container mx-auto flex flex-col p-6"
  onSubmit={[Function]}
>
  <label
    className="mb-2"
    htmlFor="comment"
  >
    Comment
  </label>
  <textarea
    className="shadow rounded border mb-2 p-2"
    id="comment"
    name="comment"
    onChange={[Function]}
    value=""
  />
  <input
    className="bg-indigo-dark hover:bg-indigo-darker text-white font-bold py-2 px-4 rounded-full"
    type="submit"
    value="Comment"
  />
</form>
`;
